2018-07-03 16:29:09,217 ERROR: Exception on /index [GET], [in C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py:1761]
Traceback (most recent call last):
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\_compat.py", line 35, in reraise
    raise value
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\Kevin\Documents\flask mega tutorial\app\routes.py", line 20, in index
    if form.validate_on_submit():
TypeError: validate_on_submit() missing 1 required positional argument: 'self'
2018-07-03 16:30:40,452 INFO: Microblog startup, [in C:\Users\Kevin\Documents\flask mega tutorial\app\__init__.py:58]
2018-07-03 16:31:05,856 ERROR: Exception on /index [GET], [in C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py:1761]
Traceback (most recent call last):
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\_compat.py", line 35, in reraise
    raise value
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\Kevin\Documents\flask mega tutorial\app\routes.py", line 28, in index
    return render_template('index.html', title='Home', form=form, posts=posts)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\templating.py", line 135, in render_template
    context, ctx.app)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\templating.py", line 117, in _render
    rv = template.render(context)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\Kevin\Documents\flask mega tutorial\app\templates\index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\Kevin\Documents\flask mega tutorial\app\templates\base.html", line 33, in top-level template code
    {% block content %}{% endblock %}
  File "C:\Users\Kevin\Documents\flask mega tutorial\app\templates\index.html", line 21, in block "content"
    {% include '-post.html' %}
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\templating.py", line 58, in get_source
    return self._get_source_fast(environment, template)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\templating.py", line 86, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: -post.html
2018-07-03 16:31:08,928 ERROR: Exception on /index [GET], [in C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py:1761]
Traceback (most recent call last):
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\_compat.py", line 35, in reraise
    raise value
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\Kevin\Documents\flask mega tutorial\app\routes.py", line 28, in index
    return render_template('index.html', title='Home', form=form, posts=posts)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\templating.py", line 135, in render_template
    context, ctx.app)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\templating.py", line 117, in _render
    rv = template.render(context)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\Kevin\Documents\flask mega tutorial\app\templates\index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\Kevin\Documents\flask mega tutorial\app\templates\base.html", line 33, in top-level template code
    {% block content %}{% endblock %}
  File "C:\Users\Kevin\Documents\flask mega tutorial\app\templates\index.html", line 21, in block "content"
    {% include '-post.html' %}
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\templating.py", line 58, in get_source
    return self._get_source_fast(environment, template)
  File "C:\Users\Kevin\Anaconda3\envs\flaskenv\lib\site-packages\flask\templating.py", line 86, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: -post.html
2018-07-03 16:31:39,541 INFO: Microblog startup, [in C:\Users\Kevin\Documents\flask mega tutorial\app\__init__.py:58]
